### ISSUE TITLE
Pure virtual function trampolines give terrible error messages

### ISSUE THREAD

**Issue by virtuald** [robotpy/robotpy-build#48]
Referencing bug report https://github.com/robotpy/robotpy-commands-v1/issues/1

```
RuntimeError: Tried to call pure virtual function "CxxBase::_isFinished"
```

This will give the same base for everything. Should fix that.

**Comment by auscompgeek** [comment]
We should also make them behave like Python abstract classes (i.e. when instantiating a class with missing methods, error), so we don't get these potentially confusing error messages.

**Comment by virtuald** [comment]
Fixed class name in a72bc04ea83d46eff83476b6b367d7628f54652f, but open to better implementations.

**Comment by virtuald** [comment]
Filed a bug at pybind11: https://github.com/pybind/pybind11/issues/2106
